SpatialGDK/Source/SpatialGDKFunctionalTests/SpatialGDK/EventTracingTests/EventTracingTest.cpp

```diff
@@ -70,42 +71,35 @@ void AEventTracingTest::PrepareTest()
 
 	AddStep(TEXT("WaitForTestToEnd"), WorkerDefinition, nullptr, nullptr, [this](float DeltaTime) {
 		WaitForTestToEnd();
+		FinishStep();
 	});
 
 	AddStep(
 		TEXT("GatherData"), WorkerDefinition, nullptr,
 		[this]() {
 			GatherData();
+			FinishStep();
 		},
 		nullptr);
 
 	AddStep(
 		TEXT("FinishEventTraceTest"), WorkerDefinition, nullptr,
 		[this]() {
 			FinishEventTraceTest();
+			FinishStep();
 		},
 		nullptr);
 }
 
 void AEventTracingTest::StartEventTracingTest()
 {
 	TestStartTime = FDateTime::Now();
-	FinishStep();
 }
 
 void AEventTracingTest::WaitForTestToEnd()
 {
-	if (TestStartTime + FTimespan::FromSeconds(TestTime) > FDateTime::Now())
-	{
-		return;
-	}
-
-	FinishStep();
-}
-
-void AEventTracingTest::FinishEventTraceTest()
-{
-	FinishStep();
+	const bool bTimerFinished = FDateTime::Now() > TestStartTime + FTimespan::FromSeconds(TestTime);
+	RequireEqual_Bool(bTimerFinished, true, TEXT("Waiting for test to end."));
```

Comment:
could even do compareFloat if the above are floats (not sure)

Reply:
but this is fine as well - since this require should always pass eventually

Reply:
I think they are datetimes.

Reply:
Random thought: we could make a templated `RequireEqual<>` that takes types that have the necessary operators defined?

Reply:
we could, though we try to also print them, so would have to be kind of smart, e.g. looking for a ToString function or equivalent.

Reply:
Sounds like a good idea